administrator/components/com_content/models/article.php

```diff
@@ -706,6 +705,59 @@ public function featured($pks, $value = 0)
 	}
 
 	/**
+	 * Method to store the token generated.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since 3.7
+	 */
+	public function shareTokenGenerate()
+	{
+		jimport('joomla.user.helper');
+		$token = JUserHelper::genRandomPassword(16);
+
+		return $token;
+	}
+
+	/**
+	 * Method to store the token generated.
+	 *
+	 * @return  token.
+	 *
+	 * @since   3.7
+	 */
+	public function shareToken()
+	{
+		$table = $this->getTable('Share', 'ContentTable');
+		$jinput = JFactory::getApplication()->input;
+		$articleId = $jinput->get('id', 0);
+		
+        	$db = $this->getDbo();
+        	$query = $db->getQuery(true);
+
+        	$query
+             		->select('sharetoken')
+             		->from($db->quoteName('#__share_draft'))
+             		->where($db->quoteName('articleId') . '=' . $db->quote($articleId));
```

Comment:
This is an integer, so we should treat it like such using (int) instead of $db->quote